contracts/src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol

```diff
@@ -610,26 +608,18 @@ contract NoCancelVRFCoordinatorV2 is
     return payment;
   }
 
-  // Get the amount of gas used for fulfillment
+  // calculatePaymentAmount will always return _only_ the premium being charged.
   function calculatePaymentAmount(
     uint256 startGas,
     uint256 gasAfterPaymentCalculation,
     uint32 fulfillmentFlatFeeLinkPPM,
     uint256 weiPerUnitGas
   ) internal view returns (uint96) {
-    int256 weiPerUnitLink;
-    weiPerUnitLink = getFeedData();
-    if (weiPerUnitLink <= 0) {
-      revert InvalidLinkWeiPrice(weiPerUnitLink);
-    }
-    // (1e18 juels/link) (wei/gas * gas) / (wei/link) = juels
-    uint256 paymentNoFee = (1e18 * weiPerUnitGas * (gasAfterPaymentCalculation + startGas - gasleft())) /
-      uint256(weiPerUnitLink);
     uint256 fee = 1e12 * uint256(fulfillmentFlatFeeLinkPPM);
-    if (paymentNoFee > (1e27 - fee)) {
-      revert PaymentTooLarge(); // Payment + fee cannot be more than all of the link in existence.
+    if (fee > (1e27 - fee)) {
+      revert PaymentTooLarge(); // Fee cannot be more than all of the link in existence.
     }
-    return uint96(paymentNoFee + fee);
+    return uint96(fee);
```

Comment:
Should we just axe this function? Is the `if (fee > (1e27 - fee)) {` check worth the gas if `fee` is always just the premium in juels?

Reply:
I guess the validation would be more appropriate to do in `setConfig`. Also, do we know if total supply of dummy link will be 10^9?

Reply:
Yeah, I believe the LINK supply on the appchains will be the same as the main chains.

Reply:
Good note on adding validation in `setConfig` though.